plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeRenameToWithGlueMetastore.java

```diff
@@ -0,0 +1,145 @@
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_tables_with_custom_location" + randomTableSuffix();
+    protected static final String CATALOG_NAME = "delta_with_custom_location";
+    protected File metastoreDir;
+    protected HiveMetastore metastore;
+    protected HdfsEnvironment hdfsEnvironment;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session deltaLakeSession = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(SCHEMA)
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(deltaLakeSession).build();
+
+        this.metastoreDir = new File(queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake_data").toString());
+        this.metastoreDir.deleteOnExit();
+
+        queryRunner.installPlugin(new DeltaLakePlugin());
+        queryRunner.createCatalog(
+                CATALOG_NAME,
+                CONNECTOR_NAME,
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore", "glue")
+                        .put("hive.metastore.glue.region", "us-east-2")
+                        .put("hive.metastore.glue.default-warehouse-dir", metastoreDir.getPath())
+                        .buildOrThrow());
+
+        HdfsConfig hdfsConfig = new HdfsConfig();
+        hdfsEnvironment = new HdfsEnvironment(
+                new DynamicHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of()),
+                hdfsConfig,
+                new NoHdfsAuthentication());
+        GlueHiveMetastoreConfig glueConfig = new GlueHiveMetastoreConfig()
+                .setGlueRegion("us-east-2");
+        metastore = new GlueHiveMetastore(
+                hdfsEnvironment,
+                glueConfig,
+                DefaultAWSCredentialsProviderChain.getInstance(),
+                directExecutor(),
+                new DefaultGlueColumnStatisticsProviderFactory(directExecutor(), directExecutor()),
+                Optional.empty(),
+                table -> true);
+
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA + " WITH (location = '" + metastoreDir.getPath() + "')");
+        return queryRunner;
+    }
+
+    @Test
+    public void testRenameOfExternalTable()
+    {
+        String oldTable = "table_external_to_be_renamed_" + randomTableSuffix();
+        String newTable = "table_external_renamed_" + randomTableSuffix();
+        String location = metastoreDir.getAbsolutePath() + "/tableLocation/";
+        try {
+            assertQuerySucceeds(format("CREATE TABLE %s WITH (location = '%s') AS SELECT 1 as val ", oldTable, location));
+            String oldLocation = (String) computeScalar("SELECT \"$path\" FROM " + oldTable);
+            assertThat(query("SELECT val FROM " + oldTable))
+                    .matches("VALUES (1)");
```

Comment:
nit: I would use `assertQuery()` for simple query assertion at here and subsequent places.